recipe/recipe.yaml

```diff
@@ -1,156 +1,68 @@
 recipe:
   name: m2-binary-packages
-  version: "20250430"
+  version: "20250515"
 
 build:
-  number: 5
+  number: 6
   noarch: generic
   dynamic_linking:
     overlinking_behavior: ignore
 
 outputs:
   - package:
-      name: m2-pkg-config
-      version: "0.29.2.4"
+      name: m2-autoconf2.72
```

Comment:
Hi @xhochy and @isuruf ! Hope you are doing well :)
I'm hoping you can give me a quick explanation of how this works, I'm not too familiar with this feedstock. 
To my understanding, this creates a new package called `m2-autoconf2.72`. I had expected the autoconf package to continue being called `m2-autoconf` without the version number. Is this some sort of pre-staging?
But this recipe also lists another package named `m2-autoconf` but uses the previous version 2.71 https://github.com/xhochy/m2-binary-packages-feedstock/blob/538bb8cb09cb611b9aecdf5311e5ffd0ae04e5d9/recipe/recipe.yaml#L3142
It just seems a bit confusing.

Thanks!